source/time_stepping/conduction_time_step.cc

```diff
@@ -0,0 +1,128 @@
+
+            if (this->get_parameters().formulation_temperature_equation
+                == Parameters<dim>::Formulation::TemperatureEquation::reference_density_profile)
+              {
+                // Overwrite the density by the reference density coming from the
+                // adiabatic conditions as required by the formulation
+                for (unsigned int q=0; q<n_q_points; ++q)
+                  out.densities[q] = this->get_adiabatic_conditions().density(in.position[q]);
+              }
+            else if (this->get_parameters().formulation_temperature_equation
+                     == Parameters<dim>::Formulation::TemperatureEquation::real_density)
+              {
+                // use real density
+              }
+            else
+              AssertThrow(false, ExcNotImplemented());
+
+            // Evaluate thermal diffusivity at each quadrature point and
+            // calculate the corresponding conduction timestep, if applicable
+            for (unsigned int q=0; q<n_q_points; ++q)
+              {
+                const double k = out.thermal_conductivities[q];
+                const double rho = out.densities[q];
+                const double c_p = out.specific_heat[q];
+
+                Assert(rho * c_p > 0,
+                       ExcMessage ("The product of density and c_P needs to be a "
+                                   "non-negative quantity."));
+
+                const double thermal_diffusivity = k/(rho*c_p);
+
+                if (thermal_diffusivity > 0)
+                  {
+                    min_local_conduction_timestep = std::min(min_local_conduction_timestep,
+                                                             this->get_parameters().CFL_number*pow(cell->minimum_vertex_distance(),2.)
+                                                             / thermal_diffusivity);
+                  }
+              }
+          }
+
+      const double min_conduction_timestep = Utilities::MPI::min (min_local_conduction_timestep, this->get_mpi_communicator());
+
+      AssertThrow (min_conduction_timestep > 0,
+                   ExcMessage("The time step length for the each time step needs to be positive, "
+                              "but the computed step length was: " + std::to_string(min_conduction_timestep) + ". "
+                              "Please check for non-positive material properties."));
+
+      return std::make_pair(Reaction::advance, min_conduction_timestep);
+    }
+
+
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace TimeStepping
+  {
+    ASPECT_REGISTER_TIME_STEPPING_MODEL(ConductionTimeStep,
+                                        "conduction time step",
+                                        "This model computes the conduction time step as the minimum "
+                                        "over all cells of $ CFL h^2 \\cdot \\rho C_p / k$, "
+                                        "where k is the thermal conductivity.")
```

Comment:
It might be good to document the type of reaction (advance) which is returned by this model.